alpha/apps/kaltura/lib/kSimuliveUtils.php

```diff
@@ -326,7 +326,8 @@ public static function getSourceAssets($sourceEntries)
 	public static function getSourceDurations($sourceEntries, $event)
 	{
 		$durations = array();
-		$eventDuration = $event->getCalculatedEndTime() * self::SECOND_IN_MILLISECONDS - $event->getCalculatedStartTime() * self::SECOND_IN_MILLISECONDS;
+		$postEndDurationMs = $event->getPostEndEntryId() ? self::roundDuration(myEntryUtils::getEntryDuration($event->getPostEndEntryId(), true)) : 0;
+		$eventDuration = $event->getCalculatedEndTime() * self::SECOND_IN_MILLISECONDS - $event->getCalculatedStartTime() * self::SECOND_IN_MILLISECONDS - $postEndDurationMs;
```

Comment:
no, if will be included here - it will chop the durations array at the wrong place (as the aggregated duration will exceed in the wrong place).
we can include the preStart here, but if do so - we need to iterate the "sourceEntries" from the second element and then update the first element manually with the preStart duration but it will require to add more code.